homeassistant/components/image_processing/facebox_face_detect.py

```diff
@@ -0,0 +1,90 @@
+"""
+Component that will perform facial detection via a local facebox instance.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/image_processing.facebox_face_detect
+"""
+import base64
+import requests
+import logging
+import voluptuous as vol
+
+from homeassistant.core import split_entity_id
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.image_processing import (
+    PLATFORM_SCHEMA, ImageProcessingFaceEntity, CONF_SOURCE, CONF_ENTITY_ID,
+    CONF_NAME)
+from homeassistant.const import (CONF_IP_ADDRESS, CONF_PORT)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_IP_ADDRESS): cv.string,
+    vol.Required(CONF_PORT): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the classifier."""
+    entities = []
+    for camera in config[CONF_SOURCE]:
+        entities.append(FaceboxFaceDetectEntity(
+            config[CONF_IP_ADDRESS],
+            config[CONF_PORT],
+            camera[CONF_ENTITY_ID],
+            camera.get(CONF_NAME)
+        ))
+    add_devices(entities)
+
+
+class FaceboxFaceDetectEntity(ImageProcessingFaceEntity):
+    """Perform a classification via a Facebox."""
+
+    def __init__(self, ip, port, camera_entity, name=None):
+        """Init with the API key and model id"""
+        super().__init__()
+        self._url = "http://{}:{}/facebox/check".format(ip, port)
+        self._camera = camera_entity
+        if name:
+            self._name = name
+        else:
+            self._name = "Facebox {0}".format(
+                split_entity_id(camera_entity)[1])
+        self.total_faces = 0
+        self.faces = []
+
+    def process_image(self, image):
+        """Process an image."""
+        response = {}
+        try:
+            response = requests.post(
+                self._url,
+                json=self.encode_image(image),
+                timeout=30
+                ).json()
+        except requests.exceptions.ConnectionError:
+            _LOGGER.error("ConnectionError: Is Facebox running?")
+            response['success'] = False
+
+        if response['success']:
+            self.total_faces = response['facesCount']
```

Comment:
We shouldn't update these instance attributes directly but pass the faces and faces count through the `process_faces` method. That will fire an event with the face data and update the instance attributes.

Reply:
Face detect and face identify components work differently (confusingly). Face detect (this Pr) have a state which is the number of faces, whilst face identify components have a state which is a recognised face. I prefer the former so have avoided using process face. If we want this to be a face identify component I will use process_faces

Reply:
We can still use `process_faces`. It will not fire events if there's no confidence set and no confidence attribute in the face dict. Just make sure `faces` and `total` are of correct types.